internal/client/tui/screens/credentials/screenEdit.go

```diff
@@ -0,0 +1,162 @@
+// Package credentials содержит компоненты и логику для создания и редактирования учетных данных в TUI приложении.
+package credentials
+
+import (
+	"beliaev-aa/GophKeeper/internal/client/storage"
+	"beliaev-aa/GophKeeper/internal/client/tui"
+	"beliaev-aa/GophKeeper/internal/client/tui/components"
+	"beliaev-aa/GophKeeper/internal/client/tui/screens"
+	"beliaev-aa/GophKeeper/internal/client/tui/styles"
+	"beliaev-aa/GophKeeper/pkg/models"
+	"context"
+	"errors"
+	"github.com/charmbracelet/bubbles/textinput"
+	"github.com/charmbracelet/bubbletea"
+	"time"
+)
+
+const (
+	credTitle = iota
+	credMetadata
+	credLogin
+	credPassword
+)
+
+// CredentialEditScreen структура для экрана редактирования учетных данных.
+type CredentialEditScreen struct {
+	inputGroup components.InputGroup
+	secret     *models.Secret
+	storage    storage.Storage
+}
+
+type inputOpts struct {
+	placeholder string
+	charLimit   int
+	focus       bool
+}
+
+// Make создаёт новый экран CredentialEditScreen на основе переданных параметров.
+func (s CredentialEditScreen) Make(msg tui.NavigationMsg, _, _ int) (tui.TeaLike, error) {
+	return NewCredentialEditScreen(msg.Secret, msg.Storage), nil
+}
+
+// NewCredentialEditScreen создаёт и инициализирует новый экземпляр CredentialEditScreen.
+func NewCredentialEditScreen(secret *models.Secret, store storage.Storage) *CredentialEditScreen {
+	m := CredentialEditScreen{
+		secret:  secret,
+		storage: store,
+	}
+
+	inputs := make([]textinput.Model, 4)
+	inputs[credTitle] = newInput(inputOpts{placeholder: "Title", charLimit: 64})
+	inputs[credMetadata] = newInput(inputOpts{placeholder: "Metadata", charLimit: 64})
+	inputs[credLogin] = newInput(inputOpts{placeholder: "Login", charLimit: 64})
+	inputs[credPassword] = newInput(inputOpts{placeholder: "Password", charLimit: 64})
+
+	var buttons []components.Button
+	buttons = append(buttons, components.Button{Title: "[ Submit ]", Cmd: func() tea.Cmd {
+		err := m.Submit()
+		if err != nil {
+			return tui.ReportError(err)
+		} else {
+			return tui.SetBodyPane(tui.StorageBrowseScreen, tui.WithStorage(m.storage))
+		}
```

Comment:
Здесь тоже можно немного лаконичнее сделать проверку.
```suggestion
		if err := m.Submit(); err != nil {
			return tui.ReportError(err)
		} 
	        return tui.SetBodyPane(tui.StorageBrowseScreen, tui.WithStorage(m.storage))
```

Reply:
Исправил